update Amlogic to 5.3.x and various other cleanups

chewitt opened the pull request:
Kernel is bumped to 5.2.1 and I'm currently pointing linux package.mk at my github linux repo to avoid the make-work of updating commit patch sets against the default kernel. I'm still waiting on updated GX audio commits from @Kwiboo and BT appears to have broken in the 5.2 bump which is still being looked at. Hopefully these issues are resolved before the PR is merged (but IMHO it's no drama if we regress and then fix later). 

DVFS changes mean G12 devices now run at full clock (1.8GHz for S905X2 and S922X, 2.2GHz for A311D. There are some G12 device-tree name changes (that are not confirmed with upstream yet) and i'm still experimenting with a more modular approach for G12 'box' device-trees that differ only in compatible, ir-keymap and ethernet chips (also not endorsed by upstream yet). G12 devices need a round of boot-testing before this is merged .. I'm currently 4,500Km from my board collection.

cputemp now works via some experimental kernel commits that were rejected upstream but work fine until revised changes are submitted. I've added them because they silence discussion about heatsinks as users can now see their S922X/A311D devices are running cool. Odroid N2 wins the "who ships the biggest heatsink" award hands-down, but VIM3 (S922X) with no fan/heatsink and the default Khadas case appears to run LE/Kodi at 38-45ºC which isn't worth much concern. Other G12 'box' devices with bad heatsinks and low-ventilation boxes aren't much higher, still cooler than S912 and nowhere near the temps seen with RPi4 :)

Panfrost now runs in 32-bit userspace without sed hacks and patches but we still have the 'black text' issue which is annoying. Investigation with Panfrost devs has ruled out all the obvious things and they are now preparing T820 dEPQ tests to see if there are differences (regressions) from T860 (RK3399) which is what most development is being done with.

I'd like an explanation for why `KERNEL_UIMAGE_ENTRYADDR="0x1980000"` and the related change to `KERNEL_UIMAGE_LOADADDR="0x1980000"` are needed (without them I can't get things to boot). It's not something I understand. NB: I have very limited time at the moment and I am not looking to reopen the uImage vs. text-offset hack debate again. 

U-boot is bumped to 2019.07 for Amlogic only (can be changed to default option if Allwinner is okay with the bump) and supports Odroid N2 on mainline u-boot via WIP patches from @superna9999 that need a bit of cleanup before upstream submission - which is likely to happen once DVFS changes are merged and updated device-trees that impact on G12 support are sync'd to u-boot.

I've dropped global support for 'vendor' u-boot from scripts/mkimage as (AFAICT) nothing uses this now we switched to the UBOOT_SYSTEM 'box' approach everywhere. The Amlogic mkimage script also allows more selective content to be placed on the installer image, e.g. UBOOT_SYSTEM board devices that use extlinux only have the extlinux files, and 'box' devices only include the uEnv.ini and autoscript files.

Kodi and FFMpeg patches are tidied up for the 18.3 bump. The PR branch also contains a couple of TEMP commits that will be dropped before merge.

150balbes commented:
Do you want to dump all DTBS into the root of the FAT partition ?

MilhouseVH commented:
I ran `make oldconfig` on this PR and got this diff - the Qualcomm changes (`CONFIG_QCOM_SPMI_*` etc.) needed to be answered for the build to proceed: http://ix.io/1OLQ

chewitt commented:
kernel defconfig updated (also includes options needed for iwd) and project options updated to use iwd instead of wpa_supplicant and swap from meson-firmware to kernel-firmware.

CvH commented on packages/linux/package.mk:
```diff
@@ -16,6 +16,12 @@ PKG_STAMP="$KERNEL_TARGET $KERNEL_MAKE_EXTRACMD"
 PKG_PATCH_DIRS="$LINUX"
 
 case "$LINUX" in
+  amlogic-5.2)
```
```suggestion
  amlogic)
```

chewitt replied:
I'm already on 5.3-pre-rc1 so for me 5.2 is stoneage :) .. and because so many different kernels are being used I prefer to keep the numbering obvious until we drop the separate kernel tree and use upstream (5.4 earliest).

CvH replied:
oki no problem :)

CvH commented on projects/Amlogic/options:
```diff
@@ -9,7 +9,7 @@
 
   # Linux Kernel to use
-    LINUX="default"
+    LINUX="amlogic-5.2"
```
```suggestion
    LINUX="amlogic"
```

chewitt replied:
it adds a tiny complication to the merged branch but helps keep things simple when you're hopping between 5.1/5.2/5.3 branches to test things.

lrusak commented on packages/tools/u-boot/package.mk:
```diff
@@ -21,6 +21,12 @@ PKG_NEED_UNPACK="$PROJECT_DIR/$PROJECT/bootloader"
 
 case "$PROJECT" in
+  Amlogic)
+    PKG_VERSION="2019.07"
```
Just bump the u-boot version

CvH replied:
could be done in a follow up pr too i guess (as chewitt is currently short on time)

chewitt replied:
do we want experimental (Amlogic) patches applied to all u-boot things? .. or we keep Amlogic separate?

lrusak replied:
The patch looks fine to me to apply globally

lrusak commented:
>  project options updated to use iwd instead of wpa_supplicant

I wouldn't do that project specifically. It should be builders choice until the switch can be made globally.

lrusak commented:
Please squash logical commits

kodi patch changes + kodi patch drops

linux kernel bumps + kernel config changes

etc

chewitt commented:
u-boot bumped for all, iwd removed (but kernel config still supports it), and BT issue resolved. Still some open questions that need answers..

lrusak commented:
> Still some open questions that need answers..

Like what?

5schatten commented on projects/Amlogic/bootloader/mkimage:
```diff
@@ -3,28 +3,61 @@
+mkimage_uEnv(){
+  echo "image: creating uEnv.ini"
 cat << EOF > "${LE_TMP}/uEnv.ini"
-dtb_name=/dtb/${DTB}
```
Is there any reason why the dtb files were moved to the root instead of keeping them in a subfolder? Keeping them in the dtb folder was IMHO a somewhat cleaner approach :thinking:

clementperon commented:
Just to point out that the u-boot bump will brick Allwinner @jernejsk

lrusak commented:
> Just to point out that the u-boot bump will brick Allwinner @jernejsk

Why do you say that?

clementperon commented:
> > Just to point out that the u-boot bump will brick Allwinner @jernejsk
> 
> Why do you say that?

some H6 patches are not applicable

lrusak commented:
> > > Just to point out that the u-boot bump will brick Allwinner @jernejsk
> > 
> > 
> > Why do you say that?
> 
> some H6 patches are not applicable

That's fine. Our normal update tar files don't update uboot anyways. Only the board specific tar files or IMG files include a u-boot update.

@jernejsk can manage but thanks for the heads-up 👍

jernejsk commented on packages/tools/u-boot/package.mk:
```diff
@@ -28,8 +28,8 @@ case "$PROJECT" in
-    PKG_VERSION="2019.04"
-    PKG_SHA256="76b7772d156b3ddd7644c8a1736081e55b78828537ff714065d21dbade229bef"
+    PKG_VERSION="2019.07"
+    PKG_SHA256="bff4fa77e8da17521c030ca4c5b947a056c1b1be4d3e6ee8637020b8d50251d0"
```
This change can be dropped now, it's in master already: https://github.com/LibreELEC/LibreELEC.tv/pull/3756

MilhouseVH commented on projects/Amlogic/filesystem/usr/bin/cputemp:
```diff
@@ -3,6 +3,14 @@
 # Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
 
-TEMP=$(cat /sys/class/hwmon/hwmon0/temp1_input)
+case `dtsoc` in
```
```suggestion
case $(dtsoc) in
```

MilhouseVH commented on projects/Amlogic/bootloader/mkimage:
```diff
@@ -3,28 +3,61 @@
 
 . config/options $1
```
```suggestion
. config/options ""
```
as arg1 will be the arg1 passed to `scripts/mkimage ` (currently nothing, but probably not what you want here).

chewitt commented:
It's time this got road-tested so that wider issues get flagged. At the moment this points add-ons to my test repo, but once Jenkins is confirmed working and an official repo has populated i'll send another PR to drop that config. I've added `emmctool` which is a small helper script for dealing with the detection of emmc modules on LePotato and LaFrite boards where emmc modules need to be installed and detected after boot (else you always boot into the Android image pre-installed on the module). I also added an ffmpeg patch to reduce some log spam (for all projects). Otherwise nothing really changed since draft reviews apart from kernel/mesa versions. Panfrost still has the black text issue, but that's continuity of current state not a regression. Bombs away..

150balbes commented:
> I've added `emmctool` which is a small helper script for dealing with the detection of emmc modules on LePotato and LaFrite boards where emmc modules need to be installed and detected after boot (else you always boot into the Android image pre-installed on the module).

Can you show "printenv" output from UART console when eMMC is connected (show ENV variables from Android u-boot)?

IMHO connection of the EMMC module "on hot" (in the working system) is potentially dangerous.

chewitt commented:
@150balbes the LePotato is engineered to boot eMMC before SD and the Android u-boot env sees Android not Linux (unless you stop boot via UART, which 99% of users won't be able to do). It's possible to use `pyamlboot` to boot and flash a board with the module pre-attached (similar to the Amlogic burning tool) or you can attach the module post-boot. Da Xue (who runs LibreComputer) assures me post-boot attach is electrically safe and supported - the rebind trick is taken from a script shipped with their official Ubuntu images. If you're a recent MacBook user like me, none of the official Apple USB-C to USB adaptors work under Linux in a VM and `pyamboot` only runs in Linux, so post-boot attach is the only option. LaFrite is different as it has a recent u-boot in the SPI flash, but u-boot has the same default env that will boot Android from eMMC before checking USB ports (LaFrite uses USB not SD) so you end up in the same situation. NB: LibreComputer plans to make a cheap a USB carrier board (similar to HK and Pine64) so users can flash eMMC modules the same as an SD card.

150balbes commented:
Why use crutches, if there is a long-tested on all TV boxes with AML and a well-working solution ?
